Generate storage enums with abice/go-enum

jraddaoui opened the pull request:
Refs #1117.

djjuhasz commented:
Adding `internal/storage/types` to the [Codecov ignore list](https://github.com/artefactual-sdps/enduro/blob/main/.codecov.yml#L21) should bring the test coverage up to the minimum threshold (or close).

jraddaoui commented:
But there is an existing test in that package, I'll create a new enums package to fully ignore it in the tests.

djjuhasz commented:
@jraddaoui other than the code coverage issue, this looks good to me.

jraddaoui commented:
Still complaining about the diff, I'll add `[skip-codecov]` before merging.

jraddaoui committed:
Generate storage enums with abice/go-enum

Move them to their own package and ignore enums packages in tests.

[skip-codecov]
